halo2_proofs/src/poly/query.rs

```diff
@@ -79,22 +93,31 @@ impl<'com, C: CurveAffine, M: MSM<C>> VerifierQuery<'com, C, M> {
 }
 
 /// A polynomial query at a point
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct VerifierQuery<'com, C: CurveAffine, M: MSM<C>> {
-    /// point at which polynomial is queried
+    /// Point at which polynomial is queried
     pub(crate) point: C::Scalar,
-    /// commitment to polynomial
+    /// Commitment to polynomial
     pub(crate) commitment: CommitmentReference<'com, C, M>,
-    /// evaluation of polynomial at query point
+    /// Evaluation of polynomial at query point
     pub(crate) eval: C::Scalar,
 }
 
-impl<'com, C: CurveAffine, M: MSM<C>> Clone for VerifierQuery<'com, C, M> {
```

Comment:
This is a minor fix, I think `#[derive(Clone)]` should work, right?

Reply:
I think `Copy` should also be derived? As otherwise, we're always forced to clone or pass by ref. And we might want to copy.

If it doesn't work, leave the derive anyways. As is less code to review/manage! :)

Reply:
Gonna add `Copy` as well, if it doesn't harm the security or its design.